src/shared/sam/activation.ts

```diff
@@ -27,13 +30,55 @@ import { SamDebugConfigProvider } from './debugger/awsSamDebugger'
 import { addSamDebugConfiguration } from './debugger/commands/addSamDebugConfiguration'
 import { SamDebugSession } from './debugger/samDebugSession'
 import { AWS_SAM_DEBUG_TYPE } from './debugger/awsSamDebugConfiguration'
+import { ext } from '../extensionGlobals'
+
+const YAML_PLUGIN = 'redhat.vscode-yaml'
+const YAML_PROMPT = 'dontShowYamlPromptEver'
```

Comment:
`suppressYamlInstallationPrompt` would be a more reasonable state name

Reply:
Consider YAML_PROMPT -> STATE_NAME_SUPPRESS_YAML_PROMPT

Reply:
Works for me; as far as I know, we haven't put too much thought into these since global storage that this uses is a key:value pair that isn't surfaced anywhere else (I don't think this even shows up in a JSON file). Took a quick look through and we have zero consistency elsewhere in our codebase.

The whole thing is once we ship it, we can't change it since this is globally constant for all VS Code installations, so if you want to make a stand, now's your chance :)